drivers/rtc/rtc_emul.c

```diff
@@ -245,27 +238,22 @@ static void rtc_emul_invoke_update_callback(const struct device *dev)
 }
 #endif /* CONFIG_RTC_UPDATE */
 
-static void rtc_emul_update(struct k_work *work)
+static void rtc_emul_update(struct k_timer *timer)
 {
-	struct rtc_emul_work_delayable *work_delayable = (struct rtc_emul_work_delayable *)work;
-	const struct device *dev = work_delayable->dev;
+	const struct device *dev = timer->user_data;
 	struct rtc_emul_data *data = (struct rtc_emul_data *)dev->data;
 
-	k_work_schedule(&work_delayable->dwork, K_MSEC(1000));
-
-	k_mutex_lock(&data->lock, K_FOREVER);
-
-	rtc_emul_increment_tm(&data->datetime);
+	K_SPINLOCK_GUARDED_SECTION(&data->lock) {
+		rtc_emul_increment_tm(&data->datetime);
 
 #ifdef CONFIG_RTC_ALARM
-	rtc_emul_test_alarms(dev);
+		rtc_emul_test_alarms(dev);
 #endif /* CONFIG_RTC_ALARM */
 
 #ifdef CONFIG_RTC_UPDATE
-	rtc_emul_invoke_update_callback(dev);
+		rtc_emul_invoke_update_callback(dev);
```

Comment:
Same, please remove formatting changes.

Reply:
Not ignored just resolved as a duplicate - will fix depending on the outcome of your previous comment.